tests/cpp/test_gpu_tensorcore.cpp

```diff
@@ -1671,175 +1671,215 @@ TEST_F(GPUTTensorCoreTest, FusionAmpereViewMatmulTN_CUDA) {
+  auto scheduleCompileAndRun = [](bool use_mkn_dim_order) {
+    NVFUSER_TEST_CUDA_ARCH_GUARD(8, 0);
 
-  // Schedule gmem read and smem write:
-  //
-  // [Mo,Ko,M,K]
-  // Swizzle tv0: 128 x 32 tile:
-  tv0cw->split(-2, 8);
-  tv0cw->split(-2, 2);
-  tv0cw->split(-1, 8);
-  //        -5   -4 -3 -2 -1
-  // [Mo,Ko,Mo16,M4,M2,Ko4,K8]
-  tv0cw->swizzle(Swizzle2DType::XOR, -4, -2);
-  tv0cw->merge(-4);
-  tv0cw->merge(-3);
-  //         -3   -2  -1
-  // [Mo,Ko,Mo16,warp,K8]
-  tv0cw->split(-3, 4);
-  tv0cw->split(-3, 2);
-  //             -4  -3   -2  -1
-  // [Mo,Ko, S4, wz2, wy2, warp,K8]
-  tv0cw->axis(-4)->parallelize(ParallelType::TIDz);
-  tv0cw->axis(-3)->parallelize(ParallelType::TIDy);
-  tv0cw->axis(-2)->parallelize(ParallelType::TIDx);
-  tv0cw->axis(-1)->parallelize(ParallelType::Vectorize);
+    Fusion fusion;
+    FusionGuard fg(&fusion);
 
-  tv0cw->setMemoryType(MemoryType::Shared);
-  // [Mo,Ko,i,wy,wx,v]
+    int M = 257, N = 511, K = 136;
 
-  // [No,Ko,N,K]
-  // Swizzle tv0: 128 x 32 tile:
-  tv1cw->split(-2, 8);
-  tv1cw->split(-2, 2);
-  tv1cw->split(-1, 8);
-  //        -5   -4 -3 -2 -1
-  // [No,Ko,No16,N4,N2,Ko4,K8]
-  tv1cw->swizzle(Swizzle2DType::XOR, -4, -2);
-  tv1cw->merge(-4);
-  tv1cw->merge(-3);
-  //         -3   -2  -1
-  // [No,Ko,No16,warp,K8]
-  tv1cw->split(-3, 4);
-  tv1cw->split(-3, 2);
-  //             -4  -3   -2  -1
-  // [No,Ko, S4, wz2, wy2, warp,K8]
-  tv1cw->axis(-4)->parallelize(ParallelType::TIDz);
-  tv1cw->axis(-3)->parallelize(ParallelType::TIDy);
-  tv1cw->axis(-2)->parallelize(ParallelType::TIDx);
-  tv1cw->axis(-1)->parallelize(ParallelType::Vectorize);
+    MatMulTileOptions gemm_tile;
+    gemm_tile.cta_tile = GemmTile(128, 128, 32);
+    gemm_tile.warp_tile = GemmTile(64, 64, 32);
+    gemm_tile.instruction_tile = GemmTile(16, 8, 16);
 
-  tv1cw->setMemoryType(MemoryType::Shared);
-  // Schedule mma input
-  tv0cr->applyMmaSwizzle(MmaOperand::A);
+    // [M,K]
+    auto tv0 = makeContigTensor(2, DataType::Half);
+    // [N,K] or [K, N] if use_mkn_dim_order is true
+    auto tv1 = makeContigTensor(2, DataType::Half);
+    // Propagate the allocation domain post-broadcast.
+    if (use_mkn_dim_order) {
+      tv1->setAllocationDomain({tv1->axis(1), tv1->axis(0)}, true);
+    }
+    fusion.addInput(tv0);
+    fusion.addInput(tv1);
 
-  // [... Mi, Ni, Ki]
-  tv0b->reorder({{-2, -3}, {-3, -2}});
-  tv0b->applyMmaSwizzle(MmaOperand::A);
+    // [M,N,K] or [M, K, N] if use_mkn_dim_order is true
+    auto tv0b = use_mkn_dim_order ? broadcast(tv0, {false, false, true})
+                                  : broadcast(tv0, {false, true, false});
+    auto tv1b = broadcast(tv1, {true, false, false});
 
-  tv1cr->applyMmaSwizzle(MmaOperand::B);
-  tv1b->applyMmaSwizzle(MmaOperand::B);
+    // propagate the allocation domain in case there's one.
+    if (use_mkn_dim_order) {
+      tv1b->setAllocationDomain(
+          {tv1b->axis(0), tv1b->axis(2), tv1b->axis(1)}, true);
```

Comment:
Are we able to ignore allocation domain from this point forward and it propagates correctly? Would you mind posting the final `fusion->printTransforms()` and comparing to that in the original test?

Reply:
I am not quite sure what you mean here. 
We create a permutation vector based on the allocation domain for `tv1b` and use it to reorder the output of the mma op. The changes to the output are then propagated back up the fusion.

Reply:
What does this allocation domain supposed to do? how should we check it in the test?